packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js

```diff
@@ -85,7 +85,7 @@ export default function useAutocomplete(props) {
     filterSelectedOptions = false,
     freeSolo = false,
     getOptionDisabled,
-    getOptionLabel: getOptionLabelProp = (option) => option,
+    getOptionLabel: getOptionLabelProp = (option) => option.label || option,
```

Comment:
```suggestion
    getOptionLabel: getOptionLabelProp = (option) => option.label ?? option,
```

Otherwise `{ label: "" }` or `{ label: 0 }`will return `'[object Object]'`